### PR TITLE
Properly format sharing datepicker locale

### PR DESCRIPTION
**Pull request by skjnldsv**
Wrongly formatted. See https://github.com/mengxiong10/vue2-datepicker/blob/v3.10.4/locale.md

(Note the header strings that remains untranslated as well as the first day starting on Sun instead of Mon)
| Before | After |
|:---------:|:------:|
|image|image|

### CODE REVIEW

**Comment by skjnldsv** [comment]
Maybe this should be in the library by default?

**Review by nickvergessen — APPROVED** [review]

File: apps/files_sharing/src/mixins/SharesMixin.js [hunk 1]
```diff
@@ -110,30 +110,24 @@ export default {
...
+		// Datepicker language
 		lang() {
-			// fallback to default in case of unavailable data
+			const weekdaysShort = window.dayNamesShort
+				? window.dayNamesShort // provided by nextcloud
+				: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.']
+			const monthsShort = window.monthNamesShort
+				? window.monthNamesShort // provided by nextcloud
+				: ['Jan.', 'Feb.', 'Mar.', 'Apr.', 'May.', 'Jun.', 'Jul.', 'Aug.', 'Sep.', 'Oct.', 'Nov.', 'Dec.']
+			const firstDayOfWeek = window.firstDay ? window.firstDay : 0
```
Comment (Pytal) [inline apps/files_sharing/src/mixins/SharesMixin.js:121]:
```suggestion
			const firstDayOfWeek = window.firstDay || 0
```
Bit more concise :)

**Review by Pytal — APPROVED** [review]

File: apps/files_sharing/src/mixins/SharesMixin.js (hunk 1 above)
Comment (skjnldsv) [reply apps/files_sharing/src/mixins/SharesMixin.js:121]: Will be part of the next vue release
https://github.com/nextcloud/nextcloud-vue/pull/2397

So I'd save us some ci and merge as is :)

**Pull request merged**